Switch code coverage reporting to codecov.io

myabc opened the pull request.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://myabc-feature-codecov-coverage.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://myabc-feature-codecov-coverage.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

myabc committed:
Switch code coverage reporting to codecov.io

Codecov supports merging of coverage data from multiple builds:
https://github.com/codeclimate/ruby-test-reporter/issues/49

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Centralise code coverage config in .simplecov

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tell SimpleCov to use rails profile

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Enable SimpleCov code coverage analysis for cukes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add codecov.io badge to README

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

linki commented:
https://codecov.io/github/opf/openproject?branch=release/4.1

linki commented:
:+1:
